qemu/tests/thin_provisioning.py

```diff
@@ -0,0 +1,169 @@
+import re
+import os
+import logging
+
+from autotest.client.shared import error
+from autotest.client.shared import utils
+from autotest.client import os_dep
+from virttest import env_process
+from avocado.utils import process
+from avocado.core import exceptions
+
+
+def get_scsi_disk(session=None):
+    """
+    Get latest scsi disk which emulated by scsi_debug module.
+
+    :param session: Guest session
+    :return: Scsi id and the disk name
+    """
+    if session is None:
+        scsi_disk_info = process.system_output("lsscsi").splitlines()
+        scsi_debug = [_ for _ in scsi_disk_info if 'scsi_debug' in _][-1]
+    else:
+        scsi_disk_info = session.cmd_output('lsscsi').splitlines()
+        scsi_debug = [_ for _ in scsi_disk_info][-1]
+    scsi_debug = scsi_debug.split()
+    scsi_id = scsi_debug[0][1:-1]
+    device_name = scsi_debug[-1]
+    return scsi_id, device_name
+
+
+def get_provisioning_mode(pci_id, device, session=None):
+    """
+    Get disk provisioning_mode, value usually is 'writesame_16', depends
+    on params for scsi_debug module.
+
+    :param pci_id: Pci slot number
+    :param device: The disk name
+    :param session: Guest session
+    :return:
+    """
+    device_name = os.path.basename(device)
+    path = "/sys/block/{:s}/device/scsi_disk".format(device_name)
+    path += "/{:s}/provisioning_mode".format(pci_id)
+    try:
+        if session is None:
+            return utils.read_one_line(path).strip()
+        else:
+            session.cmd("read < {:s}".format(path))
+            return session.cmd_output("echo $REPLY")
+    except IOError:
+        error.context("Can't get the provisioning_mode", logging.info)
+
+
+def get_allocation_bitmap():
+    """
+    Get block allocation bitmap
+    """
+    path = "/sys/bus/pseudo/drivers/scsi_debug/map"
+    try:
+        return utils.read_one_line(path).strip()
+    except IOError:
+        logging.error("block allocation bitmap not exists")
+    return ""
+
+
+def get_guest_discard_disk(session):
+    """
+    Get disk without partitions in guest.
+
+    :param session: Guest session
+    :return: Guest disk name
+    """
+    list_disk_cmd = "ls /dev/[shv]d*|sed 's/[0-9]//p'|uniq -u"
+    disk = session.cmd_output(list_disk_cmd).splitlines()[0]
+    return disk
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+      'thin-provisioning' functions test using scsi_debug:
+      1) Boot up the guest
+      2) Create a file in the guest disk
+      3) Get the Bitmap and check it
+      4) Format the disk
+      5) Get the Bitmap and check it
+
+      :param test: QEMU test object
+      :param params: Dictionary with the test parameters
+      :param env: Dictionary with test environment.
+      """
+
+    # Destroy all vms to avoid emulated disk marked drity before start test
+    for vm in env.get_all_vms():
+        if vm:
+            vm.destroy()
+            env.unregister_vm(vm.name)
```

Comment:
Can write these lines above as a func() in thin_provisioning.py and add it to a class.

No need to write them every time for all cases.